bluepysnap/edges.py

```diff
@@ -27,8 +26,353 @@
+
+        Returns:
+            List of edge IDs, if ``properties`` is None;
+            Pandas Series indexed by edge IDs if ``properties`` is string;
+            Pandas DataFrame indexed by edge IDs if ``properties`` is list.
+        """
+        return self.pathway_edges(
+            source=source_node_id, target=target_node_id, properties=properties
+        )
+
+    @staticmethod
+    def _add_circuit_ids(its, source, target):
+        """Generator comprehension adding the CircuitIds to the iterator.
+
+        Notes:
+            Using closures or lambda functions would result in override functions and so the
+            source and target would be the same for all the populations.
+        """
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1]), it[2]) for it in its)
+
+    @staticmethod
+    def _add_edge_ids(its, source, target, pop_name):
+        """Generator comprehension adding the CircuitIds to the iterator."""
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1]),
+                 CircuitEdgeIds.from_dict({pop_name: it[2]})) for it in its)
+
+    @staticmethod
+    def _omit_edge_count(its, source, target):
+        """Generator comprehension adding the CircuitIds to the iterator."""
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1])) for it in its)
+
+    def iter_connections(
+            self, source=None, target=None, return_edge_ids=False,
+            return_edge_count=False):
+        """Iterate through ``source`` -> ``target`` connections.
+
+        Args:
+            source (CircuitNodeIds/int/sequence/str/mapping/None): source node group
+            target (CircuitNodeIds/int/sequence/str/mapping/None): target node group
+            return_edge_count: if True, edge count is added to yield result
+            return_edge_ids: if True, edge ID list is added to yield result
+
+        ``return_edge_count`` and ``return_edge_ids`` are mutually exclusive.
+
+        Yields:
+            (source_node_id, target_node_id, edge_ids) if return_edge_ids == True;
+            (source_node_id, target_node_id, edge_count) if return_edge_count == True;
+            (source_node_id, target_node_id) otherwise.
+        """
+        its = []
+        for name, pop in self.items():
+            it = pop.iter_connections(source=source, target=target,
+                                      return_edge_ids=return_edge_ids,
+                                      return_edge_count=return_edge_count)
+            source_pop = pop.source.name
+            target_pop = pop.target.name
+            if return_edge_count:
+                its.append(self._add_circuit_ids(it, source_pop, target_pop))
+            elif return_edge_ids:
+                its.append(self._add_edge_ids(it, source_pop, target_pop, name))
+            else:
+                its.append(self._omit_edge_count(it, source_pop, target_pop))
+        return chain.from_iterable(its)
```

Comment:
I **think** that you can do this. I have never used it though:

```suggestion
        for name, pop in self.items():
            it = pop.iter_connections(source=source, target=target,
                                      return_edge_ids=return_edge_ids,
                                      return_edge_count=return_edge_count)
            source_pop = pop.source.name
            target_pop = pop.target.name
            if return_edge_count:
                yield from self._add_circuit_ids(it, source_pop, target_pop)
            elif return_edge_ids:
                yield from self._add_edge_ids(it, source_pop, target_pop, name)
            else:
                yield from self._omit_edge_count(it, source_pop, target_pop)
```

Reply:
Did not know this syntax at all ! 
I tried and it works